src/shogun/evaluation/TimeSeriesSplitting.cpp

```diff
@@ -99,6 +100,9 @@ void CTimeSeriesSplitting::set_min_subset_size(index_t min_size)
 	    "subsets and labels.",
 	    num_labels - (num_subsets - 1) * (num_labels / num_subsets) - 1);
 	m_min_subset_size = min_size;
+
+	// Rebuild subsets considering the new minimum subset size
```

Comment:
so it is built twice?

Reply:
In init() `m_min_subset_size` is set to 1, which is then used by build_subsets() (called by the constructor). If set_min_subset_size() is called, we should rebuild the subsets. I've found this due to a failure of `timeseries_subset_linear_splits` unit test.

Reply:
Let's try to avoid this. Maybe you can shift things around that it is only built ones .. ?

Reply:
One way to achieve this would be to allow min_subset_size to be passed as an argument in the constructor (default value 1) and remove the method set_min_subset_size(). Do you think that would be a problem?